osu.Game/Screens/Select/CarouselContainer.cs

```diff
@@ -191,6 +192,58 @@ public void SelectGroup(BeatmapGroup group, BeatmapPanel panel, bool animated =
             ScrollTo(selectedY, animated);
         }
 
+        public void Sort(FilterControl.SortMode mode) {
+            switch (mode) { 
+                case FilterControl.SortMode.Artist:
+                    groups.Sort((x, y) => string.Compare(x.BeatmapSet.Metadata.Artist, y.BeatmapSet.Metadata.Artist));
+                    break;
+                case FilterControl.SortMode.Title:
+                    groups.Sort((x, y) => string.Compare(x.BeatmapSet.Metadata.Title, y.BeatmapSet.Metadata.Title));
+                    break;
+                case FilterControl.SortMode.Author:
+                    groups.Sort((x, y) => string.Compare(x.BeatmapSet.Metadata.Author, y.BeatmapSet.Metadata.Author));
+                    break;
+                case FilterControl.SortMode.Difficulty:
+                    groups.Sort((x, y) =>
+                    {
+                        float xAverage=0, yAverage=0;

```

Comment:
As far as I know, Average, unlike Sort, does not accept delegates. If that's the case, Average would only work with integers, doubles, etc.